src/js/actions/tools.js

```diff
@@ -419,9 +431,131 @@ define(function (require, exports) {
      * @private
      * @type {function()}
      */
-    var _toolModalStateChangedHandler;
+    var _toolModalStateChangedHandler,
+        _vectorMaskHandler;
     
     /**
+     * Change the tool's vector mode
+     * This function will both dispatch a vector_state_changed event for the store
+     * and modify the current tool in order to set its properties to mask mode. 
+     * 
+     * @param {boolean} vectorState
+     * @return {Promise}
+     */
+    var changeVectorState = function (vectorState) {
+        var flux = this.flux,
+            toolStore = flux.store("tool"),
+            prevVectorMode = toolStore.getVectorMode(),
+            vectorMode = vectorState,
```

Comment:
You never mutate `vectorMode`, or `vectorState. This is an unnecessary variable.